WearTilesKotlin/app/src/main/java/com/example/wear/tiles/CoroutinesTileService.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.wear.tiles
+
+import androidx.wear.tiles.RequestBuilders.ResourcesRequest
+import androidx.wear.tiles.RequestBuilders.TileRequest
+import androidx.wear.tiles.ResourceBuilders.Resources
+import androidx.wear.tiles.TileBuilders.Tile
+import androidx.wear.tiles.TileService
+import com.google.common.util.concurrent.ListenableFuture
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.guava.future
+
+/**
+ * Base class for a Kotlin and Coroutines friendly TileService.
+ */
+abstract class CoroutinesTileService : TileService() {
+    // For coroutines, use a custom scope we can cancel when the service is destroyed
+    private val serviceJob = Job()
+    protected val serviceScope = CoroutineScope(Dispatchers.Main + serviceJob)
+
+    override fun onTileRequest(
+        requestParams: TileRequest
+    ): ListenableFuture<Tile> = serviceScope.future {
+        tileRequest(requestParams)
+    }
+
+    abstract suspend fun tileRequest(requestParams: TileRequest): Tile
+
+    override fun onResourcesRequest(requestParams: ResourcesRequest):
+        ListenableFuture<Resources> = serviceScope.future {
+            resourcesRequest(requestParams)
+        }
+
+    open suspend fun resourcesRequest(requestParams: ResourcesRequest): Resources =
+        Resources.Builder().setVersion(FIXED_RESOURCES_VERSION).build()
```

Comment:
It doesn't looks like this default implementation is used anywhere, maybe it'd be best to just have this be an `abstract` method as well?